truffle/src/main/java/org/jruby/truffle/util/ByteList.java

```diff
@@ -866,17 +861,11 @@ public int hashCode() {
     }
 
     /**
-     * Remembers toString value, which is expensive for StringBuffer.
-     *
      * @return an ISO-8859-1 representation of the byte list
      */
     @Override
     public String toString() {
-        String decoded = this.stringValue;
-        if (decoded == null) {
-            this.stringValue = decoded = decode(bytes, begin, realSize, "ISO-8859-1");
-        }
-        return decoded;
+        return decode(bytes, begin, realSize, "ISO-8859-1");
```

Comment:
`"ISO-8859-1"` is wrong here (it should just use the ByteList encoding to decode), so we could take this opportunity to fix it as well.